cli/bpmetadata/path.go

```diff
@@ -0,0 +1,101 @@
+package bpmetadata
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+const (
+	regexExamples = ".*/(examples/.*)"
+	regexModules  = ".*/(modules/.*)"
+)
+
+func fileExists(path string) (bool, error) {
+	info, err := os.Stat(path)
+	if err != nil {
+		return false, fmt.Errorf("unable to read file at the provided path: %v", err)
+	}
+
+	if info.IsDir() {
+		return false, fmt.Errorf("provided path is a directory, need a valid file path.")
+	}
+
+	return true, nil
+}
+
+func getExamples(configPath string) ([]BlueprintMiscContent, error) {
+	re := regexp.MustCompile(regexExamples)
+	return getDirPaths(configPath, re)
+}
+
+func getModules(configPath string) ([]BlueprintMiscContent, error) {
+	re := regexp.MustCompile(regexModules)
+	return getDirPaths(configPath, re)
+}
+
+// getDirPaths traverses a given path and looks for directories
+// with TF configs while ignoring the .terraform* directories created and
+// used internally by the Terraform CLI
```

Comment:
nit: We could potentially use filepath.Walk to simplify this with a walkfunc that collects parent of any .tf file in a set.

Reply:
done